modules/services/window-managers/i3-sway/sway.nix

```diff
@@ -312,7 +312,9 @@ let
     ''
   else
     "") + "\n" + (if cfg.systemdIntegration then ''
-      exec "systemctl --user import-environment; systemctl --user start sway-session.target"
+      exec "systemctl --user import-environment DISPLAY WAYLAND_DISPLAY SWAYSOCK"
+      exec "hash dbus-update-activation-environment 2>/dev/null && dbus-update-activation-environment --systemd DISPLAY WAYLAND_DISPLAY SWAYSOCK"
+      exec "systemctl --user start sway-session.target"
```

Comment:
I don't think Sway guarantees the order of execution so you should just split the command over multiple lines with `\\n` and add a semicolon between them.

Reply:
In the official wiki (https://github.com/swaywm/sway/wiki#gtk-applications-take-20-seconds-to-start) they use different `exec`s

Reply:
Fair enough.